lib/trino-parquet/src/main/java/io/trino/parquet/writer/ParquetWriters.java

```diff
@@ -174,15 +195,36 @@ public ColumnWriter primitive(PrimitiveType primitive)
             int fieldDefinitionLevel = type.getMaxDefinitionLevel(path);
             int fieldRepetitionLevel = type.getMaxRepetitionLevel(path);
             ColumnDescriptor columnDescriptor = new ColumnDescriptor(path, primitive, fieldRepetitionLevel, fieldDefinitionLevel);
+            Optional<BloomFilter> bloomFilter = createBloomFilter(bloomFilterColumns, maxBloomFilterSize, columnDescriptor);
             Type trinoType = requireNonNull(trinoTypes.get(ImmutableList.copyOf(path)), "Trino type is null");
+            checkArgument(bloomFilter.isEmpty() || SUPPORTED_BLOOM_FILTER_TYPES.contains(trinoType), format("%s does not support bloom filters", trinoType));
             return new PrimitiveColumnWriter(
                     columnDescriptor,
-                    getValueWriter(valuesWriterFactory.newValuesWriter(columnDescriptor), trinoType, columnDescriptor.getPrimitiveType(), parquetTimeZone),
+                    getValueWriter(valuesWriterFactory.newValuesWriter(columnDescriptor, bloomFilter), trinoType, columnDescriptor.getPrimitiveType(), parquetTimeZone),
                     parquetProperties.newDefinitionLevelWriter(columnDescriptor),
                     parquetProperties.newRepetitionLevelWriter(columnDescriptor),
                     compressionCodec,
                     parquetProperties.getPageSizeThreshold(),
-                    pageValueCountLimit);
+                    pageValueCountLimit,
+                    bloomFilter);
+        }
+
+        private Optional<BloomFilter> createBloomFilter(Set<String> bloomFilterColumns, int maxBloomFilterSize, ColumnDescriptor columnDescriptor)
+        {
+            // TODO: Enable use of AdaptiveBlockSplitBloomFilter once parquet-mr 1.14.0 is released
+            String dotPath = Joiner.on('.').join(columnDescriptor.getPath());
+            if (bloomFilterColumns.contains(dotPath)) {
+                OptionalLong ndv = bloomFilterNDV;
```

Comment:
I would remove `bloomFilterNDV` as a configurable. It is rather difficult for a user to assess how many unique values will land in a parquet row group. Writer scaling makes it impossible to predict beforehand what file and row-group sizes we will end up with.
We can simplify this to something like
```
...
double DEFAULT_BLOOM_FILTER_FPP = 0.05;
int BLOOM_FILTER_EXPECTED_ENTRIES = 100_000;
...
int optimalNumOfBits = BlockSplitBloomFilter.optimalNumOfBits(BLOOM_FILTER_EXPECTED_ENTRIES, fpp);
return Optional.of(new BlockSplitBloomFilter(optimalNumOfBits / 8, maxBloomFilterSize));
```
BLOOM_FILTER_EXPECTED_ENTRIES hard-coded while Fpp and max size is configurable.